public/pages/DetectorResults/containers/AnomalyHistory.tsx

```diff
@@ -66,16 +77,78 @@ export const AnomalyHistory = (props: AnomalyHistoryProps) => {
     ANOMALY_HISTORY_TABS.FEATURE_BREAKDOWN
   );
 
+  const [isLoadingAnomalyResults, setIsLoadingAnomalyResults] = useState<
+    boolean
+  >(false);
+  const [bucketizedAnomalyResults, setBucketizedAnomalyResults] = useState<
+    Anomalies
+  >();
+  const [pureAnomalies, setPureAnomalies] = useState<AnomalyData[]>([]);
+  const [bucketizedAnomalySummary, setBucketizedAnomalySummary] = useState<
+    AnomalySummary
+  >(INITIAL_ANOMALY_SUMMARY);
+
   useEffect(() => {
-    getAnomalyResultsWithDateRange(
-      dispatch,
-      dateRange.startDate,
-      dateRange.endDate,
-      props.detector.id
-    );
+    async function getAnomalyResults() {
+      try {
+        setIsLoadingAnomalyResults(true);
+        const anomalySummaryResult = await dispatch(
+          searchES(
+            getAnomalySummaryQuery(
+              dateRange.startDate,
+              dateRange.endDate,
+              props.detector.id
+            )
+          )
+        );
+        setPureAnomalies(parsePureAnomalies(anomalySummaryResult));
+        setBucketizedAnomalySummary(parseAnomalySummary(anomalySummaryResult));
+        const result = await dispatch(
+          searchES(
+            getAnomalyResultsQuery(
+              dateRange.startDate,
+              dateRange.endDate,
+              1,
+              props.detector.id
+            )
+          )
+        );
+        setBucketizedAnomalyResults(parseAnomalyResults(result));
+        setIsLoadingAnomalyResults(false);
+      } catch (err) {
+        setIsLoadingAnomalyResults(false);
+        console.error(
+          `Failed to get anomaly results for ${props.detector.id}`,
+          err
+        );
+      }
+    }
+
+    if (
+      dateRange.endDate - dateRange.startDate >
+      get(props.detector, 'detectionInterval.period.interval', 1) *
+        MIN_IN_MILLI_SECS *
+        MAX_ANOMALIES
+    ) {
+      getAnomalyResults();
```

Comment:
maybe you can name it `getBucketizedAnomalyResults()`. You may also add comment above explaining why we need to do this if too big date range causes anomalies exceeding max number.

Reply:
Sure, will change